### ISSUE TITLE
Unaligned memory access while decoding xlog magic

### ISSUE THREAD

**Issue by rtsisyk** [tarantool/tarantool#1930]
It's not safe to dereference unaligned pointers on ARM and other platforms:

```
--- a/src/box/xlog.cc
+++ b/src/box/xlog.cc
@@ -1255,13 +1255,12 @@ xlog_fixheader_decode(struct xlog_fixheader *fixheader,
        const char *end = pos + XLOG_FIXHEADER_SIZE;

        /* Decode magic */
-       fixheader->magic = *(log_magic_t *)pos; <!-- SIGBUS here
+       fixheader->magic = load_u32(pos);
        if (fixheader->magic != row_marker &&
            fixheader->magic != zrow_marker) {
                tnt_error(XlogError, "invalid magic: 0x%x", fixheader->magic);
                return -1;
        }
        pos += sizeof(log_magic_t);

        /* Read length */
        const char *val = pos;
```

Please check all other places of code and add load_u32() calls instead.

**Comment by rtsisyk** [comment]
I've fixed this problem, need to be re-checked on ARM.

**Comment by rtsisyk** [comment]
Fixed by d02db77c2a58bd243eb2aca345c04ff9c4f87974.
Re-tested on aarch64 hardware.
